test/upload.js

```diff
@@ -1309,4 +1309,58 @@ describe("File upload", function () {
       expect(callback.calledOnce).to.be.true;
       sinon.assert.calledWith(callback, errRes, null);
     });
+
+    it("With checks option", async function () {
+        const fileOptions = {
+          ...securityParameters,
+          fileName: "test_file_name",
+          file: "test_file",
+          responseFields: "tags, customCoordinates, isPrivateFile, metadata",
+          useUniqueFileName: false,
+          checks: "'request.folder' : '/'",
+        };
+        var callback = sinon.spy();
+  
+        imagekit.upload(fileOptions, callback);
+  
+        expect(server.requests.length).to.be.equal(1);
+        await sleep();
+        successUploadResponse();
+        await sleep();
+  
+        var arg = server.requests[0].requestBody;
+        expect(arg.get("file")).to.be.equal("test_file");
+        expect(arg.get("fileName")).to.be.equal("test_file_name");
+        expect(arg.get("responseFields")).to.be.equal("tags, customCoordinates, isPrivateFile, metadata");
```

Comment:
Assert that we are sending checks and its value. If we are doing any normalisation that should be asserted as well like we do for JSON and boolean fields. Not applicable for this one I guess.